kornia/feature/disk/disk.py

```diff
@@ -0,0 +1,121 @@
+                )
+            else:
+                raise
+
+        if unet_output.shape[1] != self.desc_dim + 1:
+            raise ValueError(
+                f'U-Net output has {unet_output.shape[1]} channels, but expected self.desc_dim={self.desc_dim} + 1.'
+            )
+
+        descriptors = unet_output[:, : self.desc_dim]
+        heatmaps = unet_output[:, self.desc_dim :]
+
+        return heatmaps, descriptors
+
+    def forward(
+        self, images: Tensor, n: Optional[int] = None, window_size: int = 5, score_threshold: float = 0.0
+    ) -> List[DISKFeatures]:
+        """Detects features in an image, returning keypoint locations, descriptors and detection scores.
+
+        Args:
+            images: The image to detect features in. Shape :math:`(B, C, H, W)`.
+            n: The maximum number of keypoints to detect. If None, all keypoints are returned.
+            window_size: The size of the non-maxima suppression window used to filter detections.
+            score_threshold: The minimum score a detection must have to be returned.
+                             See :py:class:`DISKFeatures` for details.
+
+        Returns:
+            A list of length :math:`B` containing the detected features.
+        """
+        B = images.shape[0]
+        heatmaps, descriptors = self.heatmap_and_dense_descriptors(images)
+        keypoints = heatmap_to_keypoints(heatmaps, n=n, window_size=window_size, score_threshold=score_threshold)
+
+        features = []
+        for i in range(B):
+            features.append(keypoints[i].merge_with_descriptors(descriptors[i]))
+
+        return features
+
+    @classmethod
+    def from_pretrained(cls, checkpoint: str = 'depth', device: torch.device = torch.device('cpu')) -> 'DISK':
+        r"""Loads a pretrained model.
+
+        Depth model was trained using depth map supervision and is slightly more precise but biased to detect keypoints
+        only where SfM depth is available. Epipolar model was trained using epipolar geometry supervision and
+        is less precise but detects keypoints everywhere where they are matchable. The difference is especially
+        pronounced on thin structures and on edges of objects.
+
+        Args:
+            checkpoint: The checkpoint to load. One of 'depth' or 'epipolar'.
+            device: The device to load the model to.
+
+        Returns:
+            The pretrained model.
+        """
+        if checkpoint == 'depth':
+            pretrained_dict = torch.hub.load_state_dict_from_url(
+                'https://raw.githubusercontent.com/cvlab-epfl/disk/master/depth-save.pth',
+                map_location=map_location_to_cpu,
+            )
+        elif checkpoint == 'epipolar':
+            pretrained_dict = torch.hub.load_state_dict_from_url(
+                'https://raw.githubusercontent.com/cvlab-epfl/disk/master/epipolar-save.pth',
+                map_location=map_location_to_cpu,
+            )
```

Comment:
load direct to `device` here didn't work? e.g to avoid load on cpu and move to gpu